Document GHC-11861 (#88)
haskell/error-messages: message-index/messages/GHC-11861/example2/index.md

---
title: Empty character literals
summary: Incorrect TemplateHaskell syntax
introduced: 9.6.1
severity: error
---

An empty pair of single quotes is not a Haskell character literal.

When Template Haskell is enabled, `''T` is syntax that refers to a quoted name of the type `T`. In this case, `''` is not an empty character literal, but rather part of the quotation syntax for type names. Read more at [TemplateHaskellQuotes](https://downloads.haskell.org/ghc/latest/docs/html/users_guide/exts/template_haskell.html#extension-TemplateHaskellQuotes).

> Empty literal character error means in that context, that you did not put a constructor name after quotation marks.

Reviewer: ```suggestion
In this context, GHC chooses to interpret `''` as an empty character literal rather than as an incomplete quoted type name, which results in this message.
```

---
title: Empty character literals
summary: Incorrect TemplateHaskell syntax
introduced: 9.6.1
severity: error
---

> Empty character was given in a single quote. This is not a valid literal.

Reviewer: ```suggestion
An empty pair of single quotes is not a Haskell character literal.
```

If you use TemplateHaskell `''T` syntax is valid and:

> > ''T has type Name, and names the type constructor T. That is, ''⟨thing⟩ interprets ⟨thing⟩ in a type context.

Reviewer: What is this quoting? I think that block quotations should have a link to a source, but I'm not sure this needs to be a quotation.

Reviewer: What about the phrasing:

When Template Haskell is enabled, `''T` is syntax that refers to a quoted name of the type `T`. In this case, `''` is not an empty character literal, but rather part of the quotation syntax for type names.

Author: Rephrased and added link to template Haskell documentation part for TemplateHaskellQuotes.
